compiler/passes/docs.cpp

```diff
@@ -240,7 +240,9 @@ void createDocsFileFolders(std::string filename) {
     dirCutoff += total;
     std::string shorter = filename.substr(dirCutoff+1);
     std::string otherHalf = filename.substr(0, dirCutoff);
-    mkdir(otherHalf.c_str(), S_IWUSR|S_IRUSR|S_IXUSR);
+    if (otherHalf.length() > 0) {
+      makeDir(otherHalf.c_str());
+    }
```

Comment:
Yes, I think that is better, since umask will reduce the permissions accordingly. If the user actually wants rwx------, they can set their umask to 0077. #1501 changed this for the output dir. I think it was just oversight that we didn't change it for these sub-directories.